<launch>
  <arg name="robot" default="UGV1"/>
  <arg name="use_gps" default="False"/>
  <arg name="is_sim" default="True"/>
  <arg name="cmd_vel" default="/jackal_velocity_controller/cmd_vel"/>
  <arg name="cw_ccw" default="False"/>
  <group ns="$(arg robot)">
    <node name="path_node" pkg="url_uavugv" type="figure8_ugv" output="screen">
      <param name="robot" type="str" value="$(arg robot)"/>
      <param name="radius" type="double" value="3.0"/>
      <!--distance(center1_x,center2_x) = 2*radius-->
      <param name="center1_x" type="double" value="5.0"/>
      <param name="center2_x" type="double" value="-1.0"/>
      <param name="max_speed" type="double" value="1.0"/>
      <param name="look_ahead" type="double" value="0.8"/>
      <param name="laps_completed" type="int" value="1"/>
      <param name="cw_ccw" type="bool" value="$(arg cw_ccw)"/> <!--false ccw / true cw-->
      <param name="use_gps" type="bool" value="$(arg use_gps)"/>
      <param name="is_sim" type="bool" value="$(arg is_sim)"/>
      <param name="cmd_vel" type="str" value="$(arg cmd_vel)"/>
    </node>
  <node name="visualization_node" pkg="url_uavugv" type="visual" output="screen">
    <param name="robot" type="str" value="$(arg robot)"/>
    <param name="use_gps" type="bool" value="$(arg use_gps)"/>
    <param name="is_sim" type="bool" value="$(arg is_sim)"/>
    <param name="cmd_vel" type="str" value="$(arg cmd_vel)"/>
  </node>
  <node name="failsafe_node" pkg="url_uavugv" type="fail_safe" output="screen">
    <param name="robot" type="str" value="$(arg robot)"/>
    <param name="use_gps" type="bool" value="$(arg use_gps)"/>
    <param name="is_sim" type="bool" value="$(arg is_sim)"/>
    <param name="cmd_vel" type="str" value="$(arg cmd_vel)"/>
  </node>
  <node name="rviz_$(arg robot)" pkg="rviz" type="rviz" args="-d $(find url_uavugv)/rviz/check.rviz" output="screen" />
  <node pkg="tf" type="static_transform_publisher" name="world_broadcaster_$(arg robot)" args="0 0 0 0 0 0 1 world odom 10" />
  </group>
</launch>